core/primitives/src/types.rs

```diff
@@ -253,6 +295,19 @@ pub enum BlockId {
 
 pub type MaybeBlockId = Option<BlockId>;
 
+#[derive(Debug, Clone, PartialEq, Eq, BorshSerialize, BorshDeserialize, Serialize, Deserialize)]
+#[serde(rename_all = "snake_case")]
+pub enum BlockCheckpoint {
```

Comment:
Block checkpoint might already have a meaning in our system, consider `BlockIdOrFinality`. CC @SkidanovAlex

Reply:
Yeah, I have considered `BlockIdOrFinality`, but I did not like the verbosity of it. I am fine with renaming it, though.

Reply:
Renamed to `BlockIdOrFinality`